Code/Framework/AzCore/AzCore/Serialization/ObjectStream.cpp

```diff
@@ -1242,6 +1246,10 @@ namespace AZ
                     if (GenericClassInfo* genericClassInfo = sc.FindGenericClassInfo(cd->m_typeId))
                     {
                         element.m_id = genericClassInfo->GetSpecializedTypeId();
+                        if (genericClassInfo->GetGenericTypeId() == GetAssetClassId())
+                        {
+                            element.m_id = GetAssetClassId();
+                        }
```

Comment:
Repeat: The `element.m_id` should remain set to the specialized type id and the LoadClass check for `if (element.m_id == GetAssetClassId())` should be updated to check the genericClassInfo.

```
if (GenericClassInfo* genericClassInfo = m_sc->FindGenericClassInfo(element.m_id);
    genericClassInfo && genericClassInfo->GetGenericTypeId() == GetAssetClassId())
{
   // ....
```